crates/starknet/src/starknet/mod.rs

```diff
@@ -159,6 +175,95 @@ impl Starknet {
 
+        // Re-execute transactions
+        if transactions.is_some() {
+            // is_some() is need here?
+            for (_hash, transaction) in transactions.unwrap_or_default().iter() {
+                match transaction.inner.clone() {
+                    Transaction::Declare(DeclareTransaction::Version0(_tx)) => {
+                        panic!("DeclareTransactionV0V1 is not supported");
+                    }
+                    Transaction::Declare(DeclareTransaction::Version1(tx)) => {
+                        let contract_class = this
+                            .state
+                            .contract_classes
+                            .get(&tx.class_hash)
+                            .expect("Failed to load Cairo0ContractClass from state");
+                        if let ContractClass::Cairo0(contract) = contract_class {
+                            let declare_tx = BroadcastedDeclareTransactionV1::new(
+                                tx.sender_address,
+                                tx.max_fee,
+                                &tx.signature,
+                                tx.nonce,
+                                contract,
+                                tx.version,
+                            );
+                            let result = this.add_declare_transaction_v1(declare_tx);
+                            if result.is_err() {
+                                panic!("Failed to add BroadcastedDeclareTransactionV1");
+                            }
+                        } else {
+                            panic!("Failed to load Cairo0ContractClass");
+                        };
+                    }
+                    Transaction::Declare(DeclareTransaction::Version2(tx)) => {
+                        let declare_tx = BroadcastedDeclareTransactionV2::new(
+                            &tx.contract_class,
+                            tx.compiled_class_hash,
+                            tx.sender_address,
+                            tx.max_fee,
+                            &tx.signature,
+                            tx.nonce,
+                            tx.version,
+                        );
+                        let result = this.add_declare_transaction_v2(declare_tx);
+                        if result.is_err() {
+                            panic!("Failed to add BroadcastedDeclareTransactionV2");
+                        }
+                    }
+                    Transaction::DeployAccount(tx) => {
+                        let deploy_account_tx = BroadcastedDeployAccountTransaction::new(
+                            &tx.constructor_calldata,
+                            tx.max_fee,
+                            &tx.signature,
+                            tx.nonce,
+                            tx.class_hash,
+                            tx.contract_address_salt,
+                            tx.version,
+                        );
+                        println!("DeployAccount!!!!");
```

Comment:
Looks like shall be removed

Reply:
Removed in d6ceb9f587a58f47653d937c1883e6fe3d172f4f